File: medium/560. Subarray Sum Equals K/560. Subarray Sum Equals K.ts
function subarraySum(nums: number[], k: number): number {
    let res = 0
    let currSum = 0
    const ht = new Map<number, number>([[0, 1]])

    for (const num of nums) {
        currSum += num
        const subArray = currSum - k

        res += ht.get(subArray) || 0
        ht.set(currSum, 1 + (ht.get(currSum) || 0))
    }

    return res
};

File: medium/560. Subarray Sum Equals K/560. Subarray Sum Equals K.py
from collections import defaultdict
from typing import List


class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        ht = defaultdict(int)
        ht[0] = 1

        total = 0
        res = 0

        for num in nums:
            total += num
            sub_array = total - k
            
            res += ht[sub_array]
            ht[total] = 1 + ht[total]

        return res
